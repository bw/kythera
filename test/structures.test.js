const t = require("./util").test

describe("Structures (obj and list)", () => {
	t("Object access", "", {skip: true})
	t("Object insertion", "", {skip: true})
	t("Array access", `
let myAr = [1, 2, 3, 4]
let r = myAr[1]
`)
	t("Array insertion", "", {skip: true})
})