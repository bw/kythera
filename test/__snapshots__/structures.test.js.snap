// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structures (obj and list) [COMPL] Array access 1`] = `"let myAr = new KYTHERA.value([new KYTHERA.value(1, KYTHERA.type.PRIMITIVES[\\"int\\"]),new KYTHERA.value(2, KYTHERA.type.PRIMITIVES[\\"int\\"]),new KYTHERA.value(3, KYTHERA.type.PRIMITIVES[\\"int\\"]),new KYTHERA.value(4, KYTHERA.type.PRIMITIVES[\\"int\\"])], new KYTHERA.type(\\"list\\", { contains: KYTHERA.type.PRIMITIVES[\\"int\\"]}));let r = myAr.value[(new KYTHERA.value(1, KYTHERA.type.PRIMITIVES[\\"int\\"])).value];"`;

exports[`Structures (obj and list) [EVALT] Array access 1`] = `
Object {
  "myAr":  {
    "type":  {
      "baseType": "list",
      "structure": Object {
        "contains":  {
          "baseType": "int",
        },
      },
    },
    "value": Array [
       {
        "type":  {
          "baseType": "int",
        },
        "value": 1,
      },
       {
        "type":  {
          "baseType": "int",
        },
        "value": 2,
      },
       {
        "type":  {
          "baseType": "int",
        },
        "value": 3,
      },
       {
        "type":  {
          "baseType": "int",
        },
        "value": 4,
      },
    ],
  },
  "r":  {
    "type":  {
      "baseType": "int",
    },
    "value": 2,
  },
}
`;

exports[`Structures (obj and list) [PARSE] Array access 1`] = `
Array [
  ParseNode {
    "identifier": "myAr",
    "kind": "let",
    "value": ParseNode {
      "elements": Array [
        ParseNode {
          "kind": "literal",
          "type": ParseNode {
            "baseType": "int",
            "kind": "type",
            "origin": "builtin",
          },
          "value": 1,
        },
        ParseNode {
          "kind": "literal",
          "type": ParseNode {
            "baseType": "int",
            "kind": "type",
            "origin": "builtin",
          },
          "value": 2,
        },
        ParseNode {
          "kind": "literal",
          "type": ParseNode {
            "baseType": "int",
            "kind": "type",
            "origin": "builtin",
          },
          "value": 3,
        },
        ParseNode {
          "kind": "literal",
          "type": ParseNode {
            "baseType": "int",
            "kind": "type",
            "origin": "builtin",
          },
          "value": 4,
        },
      ],
      "kind": "literal",
      "type": ParseNode {
        "baseType": "list",
        "contains": ParseNode {
          "baseType": "int",
          "kind": "type",
          "origin": "builtin",
        },
        "kind": "type",
        "origin": "builtin",
      },
    },
  },
  ParseNode {
    "identifier": "r",
    "kind": "let",
    "value": ParseNode {
      "index": ParseNode {
        "kind": "literal",
        "type": ParseNode {
          "baseType": "int",
          "kind": "type",
          "origin": "builtin",
        },
        "value": 1,
      },
      "kind": "access",
      "method": "bracket",
      "target": ParseNode {
        "kind": "identifier",
        "name": "myAr",
      },
    },
  },
]
`;
